File: zenscript-code-model/src/main/java/raylras/zen/model/type/AnyType.java
package raylras.zen.model.type;

import raylras.zen.model.CompilationEnvironment;

public final class AnyType implements Type {

    public static final AnyType INSTANCE = new AnyType();

    @Override
    public String getTypeName() {
        return "any";
    }

    @Override
    public boolean isSuperclassTo(Type type) {
        return true;
    }

    @Override
    public boolean isCastableTo(Type type, CompilationEnvironment env) {
        return true;
    }

    @Override
    public String toString() {
        return getTypeName();
    }

}

File: zenscript-code-model/src/main/kotlin/raylras/zen/model/type/AnyType.kt
package raylras.zen.model.type

import raylras.zen.model.CompilationEnvironment

object AnyType : Type {
    override val typeName = "any"

    override fun isSupertypeTo(type: Type) = true

    override fun isCastableTo(that: Type?, env: CompilationEnvironment) = true

    override fun toString() = typeName
}
